lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,247 @@
+# February 2016, Rohan Gandhi Korlepara
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'feature'
+require_relative 'vdc'
+
+module Cisco
+  # node_utils class for bridge_domain
+  class BridgeDomain < NodeUtil
+    attr_reader :name, :bd_ids
+
+    def initialize(bds, instantiate=true)
+      @bd_ids = bds.to_s
+      fail 'bridge-domain value is empty' if @bd_ids.empty? # no empty strings
+      fail 'bridge-domain value has spaces' if @bd_ids.include?(' ') # no spaces
+
+      # Check if the given input bd_ids does contain only numeric value or range
+      # of numeric values
+      # Also bd_ids need to fall in range of 2-2967
+      narray = @bd_ids.split(',')
+      narray.each do |elem|
+        if elem.include?('-')
+          earray = elem.split('-')
+          earray.each do |temp_elem|
+            fail 'bridge-domain cannot be a string' unless
+                                    BridgeDomain.number?(temp_elem)
+            fail 'Invalid bridge-domain value' unless
+                                    temp_elem.to_i > 1 && temp_elem.to_i < 3968
+          end
+        else
```

Comment:
While all of this range/type checking is nice I don't know that it's really necessary or desirable. We normally try to avoid enforcing ranges because these things can differ among platforms and releases; ie. if it's out of range the cli will reject it appropriately.